Wikipedia/Code/WMFProxyServer.m

```diff
@@ -189,136 +189,122 @@ - (NSURL*)proxyURLForImageURLString:(NSString*)imageURLString {
     return [components.URL wmf_imageProxyURLWithOriginalSrc:imageURLString];
 }
 
-- (NSString*)stringByReplacingImageURLsWithProxyURLsInHTMLString:(NSString*)string {
-    static NSRegularExpression* regex;
+- (NSString*)stringByReplacingImageURLsWithProxyURLsInHTMLString:(NSString*)HTMLString targetImageWidth:(NSUInteger)targetImageWidth {
+    static NSRegularExpression* imageTagRegex;
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
-        NSString* pattern = @"(<img\\s+[^>]*src\\=)(?:\")(.*?)(?:\")(.*?[^>]*)(>)";
-        regex = [NSRegularExpression regularExpressionWithPattern:pattern
+        NSString* pattern = @"(?:<img\\s)([^>]*)(?:>)";
+        imageTagRegex = [NSRegularExpression regularExpressionWithPattern:pattern
                                                           options:NSRegularExpressionCaseInsensitive
                                                             error:nil];
     });
+    
+    //defensively copy
+    HTMLString = [HTMLString copy];
+    
+    NSMutableString *newHTMLString = [NSMutableString stringWithString:@""];
+
+    __block NSInteger location = 0;
+    [imageTagRegex enumerateMatchesInString:HTMLString options:0 range:NSMakeRange(0, HTMLString.length) usingBlock:^(NSTextCheckingResult * _Nullable imageTagResult, NSMatchingFlags flags, BOOL * _Nonnull stop) {
+        //append the next chunk that we didn't match on to the new string
+        NSString *nonMatchingStringToAppend = [HTMLString substringWithRange:NSMakeRange(location, imageTagResult.range.location - location)];
+        [newHTMLString appendString:nonMatchingStringToAppend];
+        
+        //get just the image tag contents - everything between <img and >
+        NSString *imageTagContents = [imageTagRegex replacementStringForResult:imageTagResult inString:HTMLString offset:0 template:@"$1"];
```

Comment:
It's non-optional so it seems all it could be would be an empty string meaning the img tag looked like this: `<img>`.